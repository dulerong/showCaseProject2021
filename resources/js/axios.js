import axios from 'axios'
import Vue from 'vue'
// import { store } from './store'
// import router from './router'
// import { routePath } from '~js/router/constantRoutes'
// import { apiPath } from './constantsAPI'

const baseURL = '/'

const myAPI = axios.create({
  baseURL: baseURL,
  timeout: 30000, // unit = millisecond
  headers: {
    "Content-Type": "application/json"
  }
})

// const getAuthToken = () => `Bearer ${localStorage.getItem('user-token')}`

// const authInterceptor = (config) => {
//   config.headers['Authorization'] = getAuthToken()
//   return config
// }

// myAPI.interceptors.request.use(authInterceptor)

// const errorInterceptor = err => {
//   switch(err.response.status) {
//     case 400:
//       console.log(err.response)
//       errorFourZeroZero(err)
//       break
//     case 401:
//       console.log(err.response)
//       errorFourZeroOne(err)
//     default:
//       console.log(err.response)
//       showNotification(err.response.status, '管理会社へお問合せください', 'error', -1)
//   }
//   store.commit('loading/loadingStop')
//   return Promise.reject(err)
// }

// const noAuthRedirect = () => {
//   localStorage.removeItem('user-token')
//   store.commit('auth/removeToken')
//   router.push(routePath.login.path)
// }

// const errorFourZeroZero = (err) => {
//   switch(err.response.config.url) {
//     case apiPath.customer.getDetail:
//       router.push(routePath.search.path)
//       showNotification('無効の顧客ID', '顧客検索ページに転送', 'error', -1)
//       break
//     case apiPath.auth.login:
//       showNotification('ログインIDかパスワードが間違っています', '', 'error', -1)
//       break
//     default:
//       showNotification(err.response.status, '管理会社へお問合せください', 'error', -1)
//   }
// }

// const errorFourZeroOne = (err) => {
//   switch(err.response.config.url) {
//     case apiPath.auth.changePassword:
//       break
//     default:
//       showNotification('ユーザー認証失敗', 'ログイン画面に転送', 'error', -1)
//       return noAuthRedirect()
//   }
// }

// const showNotification = (status, message, color, timeout) => {
//   store.commit('notification/showNotification', {
//     message: `エラー: ${status}, ${message}`,
//     color: color,
//     timeout: timeout
//   })
// }

// const responseInterceptor = res => {
//   switch(res.status) {
//     case 200:
//       console.log('response 200')
//       break;
//     default:
//       console.log(res)
//   }
//   return res
// }

// myAPI.interceptors.response.use(responseInterceptor, errorInterceptor)

export default myAPI

Vue.prototype.$_api = myAPI