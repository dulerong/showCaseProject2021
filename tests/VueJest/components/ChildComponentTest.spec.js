import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import ChildComponent from '~components/crm/ChildComponent'
import axios from 'axios'

jest.mock('axios')

describe('Child Component Tests', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const shallowMountFunction = options => {
    return shallowMount(ChildComponent, {
      localVue,
      vuetify,
      ...options,
    })
  }

  it('ChildComponent renders properly', () => {
    const wrapper = shallowMountFunction({
      propsData: {
        message: 'header number three is updated'
      }
    })

    expect(wrapper.exists()).toBe(true)
  })

  it('ChildComponent props confirm', () => {
    const wrapper = shallowMountFunction({
      propsData: {
        message: 'header number three is updated'
      }
    })

    expect(wrapper.props().message).toBe('header number three is updated')

    const headerThree = wrapper.find('h3')
    expect(headerThree.text()).toMatch('is updated')
  })

  it('ChildComponent props confirm', async () => {
    const clickMe = jest.spyOn(ChildComponent.methods, 'clickMe')
    const wrapper = mount(ChildComponent, {
      localVue,
      vuetify,
      propsData: {
        message: 'something I like'
      },
      data() {
        return {
          counter: 1
        }
      }
    })
    const response = {
      data: 'responseData'
    }
    axios.get.mockResolvedValue(response)

    expect(wrapper.html()).toMatchSnapshot()

    const button = wrapper.find('.v-btn')
    expect(button.exists()).toBe(true)

    const emitEvent = jest.fn()
    wrapper.vm.$on('click-event', emitEvent)
    
    await button.trigger('click')
    expect(clickMe).toHaveBeenCalled()
    expect(emitEvent).toHaveBeenCalledTimes(1)
    expect(wrapper.vm.counter).toBe(2)

    expect(wrapper.vm.webData).toBe('responseData')
  })
})
