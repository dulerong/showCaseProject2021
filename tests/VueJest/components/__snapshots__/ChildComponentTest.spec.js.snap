// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child Component Tests ChildComponent props confirm 1`] = `
<div>
  <h3>This is a child something I like</h3><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">CLICK ME</span></button>
  <h4>1</h4>
</div>
`;
